Components/BasketService.php

```diff
@@ -194,10 +201,58 @@ private function processOrderDetail(Order $order, Detail $orderDetail)
      */
     private function addArticle(Detail $orderDetail)
     {
-        $this->sBasket->sAddArticle(
+        $basketDetailId = $this->sBasket->sAddArticle(
             $orderDetail->getArticleNumber(),
             $orderDetail->getQuantity()
         );
+
+        $this->copyDetailAttributes($orderDetail->getId(), $basketDetailId);
+    }
+
+    /**
+     * Copies attributes from the supplied order detail article ID to a basket detail ID
+     *
+     * @param $basketDetailId
+     * @param $orderDetailId
+     * @throws \Zend_Db_Adapter_Exception
+     */
+    private function copyDetailAttributes($orderDetailId, $basketDetailId)
```

Comment:
Code looks good.

A few possible improvements:
* copyDetailAttributes could be a individual service, which is then injected in the BasketManager
* adding some type's on $orderDetailId, $basketDetailId would aid as well,  
Shopware 5 tends to use string and int for ids.
* Maybe add an early void return if there are not attributes  
```
$orderDetailAttributes = $orderDetailAttributesResult[0];
if (!count($orderDetailAttributes[0])){
     return;
}
```